commit 326106bf5b928e7d88540e3643ab4ede7e412413
Author: Aaron Seigo <aseigo@kde.org>
Date:   Tue Mar 5 18:05:33 2013 +0100

    if load failed ... don't accept it.
    
    prevents failed containment settings from leaving things in a worse state
    than necessary

diff --git a/plasma/view.cpp b/plasma/view.cpp
index d2a755a..84064ae 100644
--- a/plasma/view.cpp
+++ b/plasma/view.cpp
@@ -320,6 +320,12 @@ Containment *View::swapContainment(Plasma::Containment *existing, const QString
     Plasma::Corona *corona = old->corona();
     Plasma::Containment *c = corona->addContainmentDelayed(name, args);
     if (c) {
+        const bool fail = c->pluginName() != name;
+        if (fail) {
+            delete c;
+            return old;
+        }
+
         c->init();
 
         KConfigGroup oldConfig = old->config();
